src/braces.cpp

```diff
@@ -98,6 +99,42 @@ static chunk_t *mod_case_brace_add(chunk_t *cl_colon);
 static void process_if_chain(chunk_t *br_start);
 
 
+//TODO: this should take in const refs, like almost all chunk OP
+bool paren_multiline_before_brace(chunk_t *brace, c_token_t paren_t = CT_SPAREN_CLOSE)
+{
+   if (brace == nullptr
+       || (brace->type != CT_BRACE_OPEN && brace->type != CT_BRACE_CLOSE)
+       || (brace->parent_type != CT_IF
+           && brace->parent_type != CT_ELSEIF
+           && brace->parent_type != CT_FOR
+           && brace->parent_type != CT_USING_STMT
+           && brace->parent_type != CT_WHILE
+           && brace->parent_type != CT_FUNC_CLASS_DEF
+           && brace->parent_type != CT_FUNC_DEF))
+   {
+      return(false);
+   }
+
+   // find paren pair of the if/for/while/...
+   auto paren_close = chunk_get_prev_type(brace, paren_t, brace->level, scope_e::ALL);
+   auto paren_open  = chunk_skip_to_match_rev(paren_close, scope_e::ALL);
+
```

Comment:
I didn't know the "almost-always-auto" approach. But when looking a the resulting code I think this approach misses the point and almost certainly leads to bugs. As I said before reading the code gets much more difficult as the types are not clear anymore. But there are situations where it gets even worst. Imagine
`auto  var = 0`
Is this signed or unsigned. What length does var have? Only the compiler knows. The programmer might assume something different and writes broken code.

Anyway I don't want to start a concept discussion here. But me personlly I stick to the fact that it is more important that the human understands the code than the computer.